### PR TITLE
Examine work content of global binders in 'isWorkFree' 

### PR DESCRIPTION
**Pull request by martijnbastiaan**
Edit: **I'll test this on a larger codebase later this week. Please postpone merging until then.**

..and prevent new let-bindings in `flattenCallTree`, as this breaks Clash's NF.

### CODE REVIEW

**Comment by martijnbastiaan** [comment]
Do you think we should backport this to 1.0 @christiaanb ?

File: clash-lib/src/Clash/Normalize/Util.hs [hunk 1]
```diff
@@ -67,10 +69,47 @@ import           Clash.Rewrite.Types
...
+isWorkFree
+  :: Term
+  -> NormalizeSession Bool
+isWorkFree (collectArgs -> (fun,args)) = case fun of
+  Var i -> do
+    let isPoly = isPolyFunTy (varType i)
+    if isLocalId i && not isPoly then
+      pure True
+    else
+      not <$> isRecursiveBndr i
```
Comment (christiaanb) [inline clash-lib/src/Clash/Normalize/Util.hs]: We should check whether the global function is work-free

Comment (martijnbastiaan) [reply clash-lib/src/Clash/Normalize/Util.hs]: Done?

**Comment by martijnbastiaan** [comment]
I'm awfully close to "later this week" now.

**Pull request closed without merging**